### ISSUE TITLE
Add sanity check for addons

### ISSUE THREAD

**Issue by facaiy** [tensorflow/addons#68]
Since #63 has introduced sanity check for addons, can we add it for CI? 
cc @av8ramit @karmel @seanpmorgan 

command:
```bash
make sanity-check
# or
bash tools/ci_build/install/install_ci_dependency.sh --quiet
bash tools/ci_build/ci_sanity.sh --incremental
```

**Comment by facaiy** [comment]
sanity check should always pass after #64 is merged into master.

**Comment by karmel** [comment]
This should be possible.

**Comment by karmel** [comment]
Ah, wait, modulo viewing results. @av8ramit -- for results to be public, does everything have to be run through Bazel currently? That is, we can't run arbitrary scripts like this directly?

**Comment by av8ramit** [comment]
I'll try and see if we can just invoke a script and just see the results that way.

**Comment by av8ramit** [comment]
Okay, so internally we now just invoke the script. So now you can modify and add whatever you'd like to the [script](https://github.com/tensorflow/addons/blob/master/tools/ci_testing/addons_cpu.sh) and it will run in presubmit and continuous.

**Comment by facaiy** [comment]
Yeah, it works well, but I prefer to keep them (sanity-check and unit-test) independent. What do you think?

**Comment by seanpmorgan** [comment]
Agree that it's significantly easier to review if we have a separate script ran for linting type errors. @av8ramit do you think it would be possible to run one additional script like `addons_style.sh`. We can then probably fit most other things we want ran into one of our existing shell scripts.

**Comment by av8ramit** [comment]
Do you want a separate job for the linting script? Or do you want that to run as part of CPU tests with Python2/3?

**Comment by facaiy** [comment]
>  a separate job for the linting script

+1, yes. If possible, run it in the python 3 Linux CPU environment.

**Comment by facaiy** [comment]
Friendly ping @av8ramit

**Comment by karmel** [comment]
I am working on this now; sorry for the delay. Should have it ready in the next couple of days.

**Comment by karmel** [comment]
These are now running, though failing: https://source.cloud.google.com/results/invocations/1b2b8286-1722-4c51-a05e-274bad80607b/targets/tensorflow_addons%2Fubuntu%2Fcpu%2Fsanity%2Fpresubmit/log

Looks like a unicode issue; @facaiy / @seanpmorgan can you take a look, and let me know if it's in the test or something we should fix in the test env?

**Comment by facaiy** [comment]
Thanks, Karmel, I'll take a look later. By the way, it seems python 3.4?

**Comment by facaiy** [comment]
@karmel I cannot reproduce it in the tensorflow/tensorflow:nightly-custom-op image:

```bash
root@e45e92485b69:/addons# git log -n 1
commit 3bf1b5fde8a2ba8b95f746ba01e8e14206a5bbfc
Author: Kapil Sachdeva <ksachdeva17@gmail.com>
Date:   Tue Mar 19 10:53:21 2019 -0500

    FIX - Use public keras API for custom_object decorator (#97)
root@e45e92485b69:/addons# python3 tools/test/check_futures_test.py
root@e45e92485b69:/addons# echo $?
0
root@e45e92485b69:/addons# python2 tools/test/check_futures_test.py
root@e45e92485b69:/addons# echo $?
0
root@e45e92485b69:/addons# python3 --version
Python 3.4.3
root@e45e92485b69:/addons# python2 --version
Python 2.7.6
```

Could you print the path before opening it?

https://github.com/tensorflow/addons/blob/3bf1b5fde8a2ba8b95f746ba01e8e14206a5bbfc/tools/test/check_futures_test.py#L51

**Comment by facaiy** [comment]
Gently ping @karmel @av8ramit

**Comment by karmel** [comment]
Sorry— I’m OOO, but will be able to take a look next week.

**Comment by facaiy** [comment]
That's fine, please update the issue when new information becomes available :-)

**Comment by av8ramit** [comment]
Here is what I get. Do you want me to update to Python3.5? 


```
=== Sanity check step 4 of 6:  do_check_futures_test (Check that python files have certain __future__ imports) ===

Traceback (most recent call last):
  File "check_futures_test.py", line 105, in <module>
    main()
  File "check_futures_test.py", line 101, in main
    raise AssertionError('\n'.join(error_msgs))
AssertionError: Error in tf/lib/python3.4/io.py: Missing futures: absolute_import division print_function
Error in tf/lib/python3.4/types.py: Missing futures: absolute_import division print_function

...

Error in tf/bin/activate_this.py: Missing futures: absolute_import division print_function
```

**Comment by facaiy** [comment]
No, Amit, I don't think py3.5 can fix the problem. We expect that [check_futures_test.py)](https://github.com/tensorflow/addons/blob/master/tools/test/check_futures_test.py) only checks those python files of addons. It looks amazing that tf/bin/activate_this.py is in the addons directory?

**Comment by facaiy** [comment]
Amit, do you know what is the location of the tf/lib directory? Is it in the tensorflow_addons?

**Comment by av8ramit** [comment]
@facaiy yeah I think that's it. I moved the virtualenv outside of the addons directory and [it works now](https://source.cloud.google.com/results/invocations/9997a63d-d0a3-4e55-b93c-3354878be2d7/targets).

**Comment by seanpmorgan** [comment]
Thanks for the help!

**Comment by seanpmorgan** [comment]
@av8ramit Is there an issue with displaying the logs?

For example see this report:
https://source.cloud.google.com/results/invocations/bb8ac7a3-2ef6-487d-a221-e639f83cf410bmit/log

Coming from this PR:
https://github.com/tensorflow/addons/pull/102

**Comment by av8ramit** [comment]
I didn't change anything. Might have just taken a few minutes.
